File: problems/problemset/ttl-cache/Python/solution/ttl_cache.py
class TTLCache:
    def __init__(self):
        self._now = 0
        self._store: dict[str, tuple[str, int]] = {}  # key -> (value, expiry)

    def set(self, key: str, value: str, ttl: int) -> None:
        self._store[key] = (value, self._now + ttl)

    def get(self, key: str) -> str:
        entry = self._store.get(key)
        if entry is None or self._now >= entry[1]:
            return ''
        return entry[0]

    def delete(self, key: str) -> None:
        self._store.pop(key, None)

    def size(self) -> int:
        return sum(1 for _, expiry in self._store.values() if self._now < expiry)

    def tick(self, delta: int) -> None:
        self._now += delta

File: problems/problemset/ttl-cache/Rust/solution/ttl_cache.rs
use std::collections::HashMap;

pub struct TTLCache {
    now: i64,
    store: HashMap<String, (String, i64)>, // key -> (value, expiry)
}

impl TTLCache {
    pub fn new() -> Self {
        Self { now: 0, store: HashMap::new() }
    }

    pub fn set(&mut self, key: &str, value: &str, ttl: i64) {
        self.store.insert(key.to_string(), (value.to_string(), self.now + ttl));
    }

    pub fn get(&self, key: &str) -> String {
        match self.store.get(key) {
            Some((val, expiry)) if self.now < *expiry => val.clone(),
            _ => String::new(),
        }
    }

    pub fn del(&mut self, key: &str) {
        self.store.remove(key);
    }

    pub fn size(&self) -> usize {
        self.store.values().filter(|(_, expiry)| self.now < *expiry).count()
    }

    pub fn tick(&mut self, delta: i64) {
        self.now += delta;
    }
}
